Firestore/Source/Core/FSTSyncEngine.mm

```diff
@@ -307,26 +308,35 @@ - (void)transactionWithRetries:(int)retries
     workerQueue->Enqueue(
         [self, retries, workerQueue, updateCallback, resultCallback, transaction, maybe_result] {
           if (!maybe_result.ok()) {
+            if (retries > 0 && [self isRetryableTransactionError:maybe_result.status()] &&
+                !transaction->IsPermanentlyFailed()) {
+              workerQueue->VerifyIsCurrentQueue();
+              return [self transactionWithRetries:(retries - 1)
+                                      workerQueue:workerQueue
+                                   updateCallback:updateCallback
+                                   resultCallback:resultCallback];
+            }
```

Comment:
nit: I think you just copied preexisting code, but I would wrap the below two lines in "} else { ... }" so that it's obvious that they'll only be hit if the `if` condition fails.  It just makes the control flow more obvious...  

(in general I'm not a fan of "early return" except for quickly filtering out "exceptional" cases at the tops of functions)

Same for the below code.